base_classes/nyaml/NXsample_component.yaml

```diff
@@ -0,0 +1,148 @@
+    doc: |
+      The chemical formula specified using CIF conventions.
+      Abbreviated version of CIF standard:
+      
+      * Only recognized element symbols may be used.
+      * Each element symbol is followed by a 'count' number. A count of '1' may be omitted.
+      * A space or parenthesis must separate each cluster of (element symbol + count).
+      * Where a group of elements is enclosed in parentheses, the multiplier for the
+        group must follow the closing parentheses. That is, all element and group
+        multipliers are assumed to be printed as subscripted numbers.
+      * Unless the elements are ordered in a manner that corresponds to their chemical
+        structure, the order of the elements within any group or moiety depends on
+        whether or not carbon is present.
+      * If carbon is present, the order should be:
+      
+        - C, then H, then the other elements in alphabetical order of their symbol.
+        - If carbon is not present, the elements are listed purely in alphabetic order of their symbol.
+        
+      * This is the *Hill* system used by Chemical Abstracts.
+  unit_cell_abc(NX_FLOAT):
+    deprecated: |
+      Use `NXsingle_crystal` and `NXunit_cell` to describe single crystalline materials.
+    unit: NX_LENGTH
+    doc: |
+      Crystallography unit cell parameters a, b, and c
+    dimensions:
+      dim: [[1, 3]]
+  unit_cell_alphabetagamma(NX_FLOAT):
+    deprecated: |
+      Use `NXsingle_crystal` and `NXunit_cell` to describe single crystalline materials.
+    unit: NX_ANGLE
+    doc: |
+      Crystallography unit cell parameters alpha, beta, and gamma
+    dimensions:
+      dim: [[1, 3]]
+  unit_cell_volume(NX_FLOAT):
+    deprecated: |
+      Use `NXsingle_crystal` and `NXunit_cell` to describe single crystalline materials.
+    unit: NX_VOLUME
+    doc: |
+      Volume of the unit cell
+  sample_orientation(NX_FLOAT):
+    deprecated: |
+      Use `NXsingle_crystal` and `NXunit_cell` to describe single crystalline materials.
+    unit: NX_ANGLE
+    doc: |
+      This will follow the Busing and Levy convention from Acta.Crysta v22, p457 (1967)
+    dimensions:
+      rank: 1
+      dim: [[1, 3]]
+  orientation_matrix(NX_FLOAT):
+    deprecated: |
+      Use `NXsingle_crystal` and `NXunit_cell` to describe single crystalline materials.
+    doc: |
+      Orientation matrix of single crystal sample component.
+      This will follow the Busing and Levy convention from Acta.Crysta v22, p457 (1967)
+    dimensions:
+      rank: 2
+      dim: [[1, 3], [2, 3]]
+  mass(NX_FLOAT):
+    unit: NX_MASS
+    doc: |
+      Mass of sample component
+  density(NX_FLOAT):
+    unit: NX_MASS_DENSITY
+    doc: |
+      Density of sample component
+  relative_molecular_mass(NX_FLOAT):
+    unit: NX_MASS
+    doc: |
+      Relative Molecular Mass of sample component
+  description:
+    doc: |
+      Description of the sample component
+  volume_fraction(NX_FLOAT):
```

Comment:
wouldnt this need a \@depends_on attribute then to an instance of NXsample_component_set ?

Reply:
Done